chpl/chpl-service/src/main/java/gov/healthit/chpl/service/RealWorldTestingEligiblityService.java

```diff
@@ -0,0 +1,190 @@
+package gov.healthit.chpl.service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+
+import gov.healthit.chpl.activity.history.ListingActivityUtil;
+import gov.healthit.chpl.activity.history.explorer.RealWorldTestingEligibilityActivityExplorer;
+import gov.healthit.chpl.activity.history.query.RealWorldTestingEligibilityQuery;
+import gov.healthit.chpl.dao.CertifiedProductDAO;
+import gov.healthit.chpl.domain.CertificationCriterion;
+import gov.healthit.chpl.domain.CertificationStatusEvent;
+import gov.healthit.chpl.domain.CertifiedProduct;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.dto.ActivityDTO;
+import gov.healthit.chpl.dto.CertifiedProductDTO;
+import gov.healthit.chpl.entity.CertificationStatusType;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.util.DateUtil;
+
+// Each time this class is used, a new instance is required so that the memoization is threadsafe
```

Comment:
I had the same requirement of ChplHtmlEmailBuilder. I ended up with the below code at the top of the class so I could still use it as a spring component where it is referenced and not have to use **new...** May want to consider it so that the code where this class is used is more similar to the rest of our codebase.
`@Component
@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)`